api/types/swarm/swarm.go

```diff
@@ -159,6 +160,7 @@ type InitRequest struct {
 	Availability     NodeAvailability
 	DefaultAddrPool  []string
 	SubnetSize       uint32
+	DataPathPort     uint32
```

Comment:
We need to fence this feature based on API version; when using an older version of the API, but the daemon "supports" this feature, it should still be ignored; something like this in `initCluster()`;

```go
	version := httputils.VersionFromContext(ctx)
	// DataPathPort was added in API 1.40. Ignore this option on older API versions.
	if versions.LessThan(version, "1.40") {
		req.DataPathPort = 0
	}
```

Here's a patch you can apply:

```patch
diff --git a/api/server/router/swarm/cluster_routes.go b/api/server/router/swarm/cluster_routes.go
index dda2a92d91..337cfd3419 100644
--- a/api/server/router/swarm/cluster_routes.go
+++ b/api/server/router/swarm/cluster_routes.go
@@ -27,6 +27,11 @@ func (sr *swarmRouter) initCluster(ctx context.Context, w http.ResponseWriter, r
                }
                return errdefs.InvalidParameter(err)
        }
+       version := httputils.VersionFromContext(ctx)
+       // DataPathPort was added in API 1.40. Ignore this option on older API versions.
+       if versions.LessThan(version, "1.40") {
+               req.DataPathPort = 0
+       }
        nodeID, err := sr.backend.Init(req)
        if err != nil {
                logrus.Errorf("Error initializing swarm: %v", err)
```